### PR TITLE
feat(compiler-plugin): discover per-declaration hints in collectAllModulePreviews IR transform

### PR DESCRIPTION
**Pull request by TBSten**
## Summary

- redesign-hint-mechanism の T03: `collectAllModulePreviews()` の IR transform で per-declaration hint (Metro 風 V2) を依存モジュール側から発見・利用する処理を追加
- 旧モジュール集約 hint と V2 hint の両方を merge し、 `distinctPreviewsById` で dedup
- T02 (PR #170) 必須

実装:
- `HintDiscoveryV2.kt`: `me.tbsten.compose.preview.lab.hints` package 配下の `previewHint_*` 関数を `moduleDescriptor.getPackage(...).memberScope.getFunctionNames()` で全列挙 (Metro と同じパターン、 K2 で `@ObsoleteDescriptorBasedAPI` だが de facto 標準)
- `IR_EXTERNAL_DECLARATION_STUB` origin で外部 module hint のみ filter (自モジュール emit は thisModulePreviews 経由で既に列挙済み)
- `PreviewListIrBuilder.buildConcatenatedPreviewsExpr` を拡張: V2 hint は単体 CollectedPreview を return するので `add(hint())` で list に積む
- V1 `collectDependencyGetters` は手付かずで残す (T05 で削除)

設計判断:
- 独立ファイル `HintDiscoveryV2.kt` (後続のディレクトリ再編で `transformations/...` 配下に移しやすい形)
- 同 FQN cross-artifact collision は受容済み edge case (`.local/redesign-hint-mechanism/参考.md` §3)

KLIB target / Compose context 実機 invoke は次の T04 で確認。

Refs: `.local/ticket/redesign-hint-mechanism/03-consumer-side-discovery.md`
Stacked on: #170

## Test plan

- [x] `./gradlew :compiler-plugin:test` 全 green (新規 4 件 + 既存 cross-module 関連 全件)
- [x] `./gradlew ktlintCheck` pass
- [x] `./gradlew :compiler-plugin:apiCheck :core:apiCheck` pass
- [x] kctfork 2-stage compile: cross-module hint 発見・list に積む
- [x] kctfork: 複数 `@Preview` 持つ依存モジュールの hint をすべて発見
- [x] kctfork: cross-module preview の displayName / line 等 metadata が復元
- [x] kctfork: V1 + V2 で重複しても `distinctPreviewsById` で 1 個に統合

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: compiler-plugin/src/main/kotlin/me/tbsten/compose/preview/lab/compiler/ir/PreviewListIrBuilder.kt [hunk 1]
```diff
@@ -234,9 +247,10 @@ internal class PreviewListIrBuilder(
      */
     fun buildConcatenatedPreviewsExpr(builder: DeclarationIrBuilder, thisModulePreviews: IrExpression): IrExpression {
         val dependencyGetters = cachedDependencyGetters
+        val hintsV2 = cachedHintsV2
```
[the remark this answers is not in the document]
Comment (TBSten) [reply compiler-plugin/src/main/kotlin/me/tbsten/compose/preview/lab/compiler/ir/PreviewListIrBuilder.kt:250]: Resolved downstream by T05 (#173): V1 (旧モジュール集約 hint) を削除したため、 二重計算は解消されました。 現在は per-declaration hint のみが動作するため、 各 preview に対し 1 度だけ `CollectedPreview` が構築されます。

File: compiler-plugin/src/main/kotlin/me/tbsten/compose/preview/lab/compiler/ir/PreviewListIrBuilder.kt [hunk 2]
```diff
@@ -279,6 +302,15 @@ internal class PreviewListIrBuilder(
...
                 }
             }
+            // Per-declaration hint (V2): 各 hint は CollectedPreview を 1 個 return するので
+            // `add(hint())` で list に積む (V1 dep getter のような List ではない)。
+            for (hintFn in hintsV2) {
+                val hintCall = compatContext.irCall(this, hintFn.symbol, collectedPreviewType)
+                +compatContext.irCall(this, addFun, pluginContext.irBuiltIns.booleanType).apply {
+                    arguments[0] = compatContext.irGet(this@irBlock, listVar)
+                    arguments[1] = hintCall
+                }
```
[the remark this answers is not in the document]
Comment (TBSten) [reply compiler-plugin/src/main/kotlin/me/tbsten/compose/preview/lab/compiler/ir/PreviewListIrBuilder.kt:312]: ご指摘の通り、 これは redesign 全体で **未解決の TODO** として認識しており、 `.local/redesign-hint-mechanism/README.md` §3 / §7 に記録しています。

**経緯**:
- T02 で FIR 側で `@ComposePreviewLabOption(ignore = true)` を annotation argument として読み取って hint emit から除外する案を試したが、 `lazyResolveToPhase(ANNOTATION_ARGUMENTS)` 経由でも annotation 取得が安定しなかった (詳細不明、 K2 内部の resolution timing 問題と推測)
- 暫定として hint emit 自体は ignore=true でも行い、 cross-module で露出する課題は redesign 完了後の follow-up として残した

**現状の挙動 (T05 V1 削除後)**:
- 自モジュールの `collectModulePreviews()` は IR `previews` (filter 済み) 経由なので OK
- cross-module の `collectAllModulePreviews()` 経由では ignore=true の preview も list に乗る → このコメントの指摘通り

**Follow-up 対応案** (redesign 完了後の別 ticket で扱う):
- (A) FIR phase での annotation 読み取りを再調査 (生 `symbol.fir.annotations` を直接 walk する方式等)
- (B) `CollectedPreview` に `internal val ignored: Boolean` field を追加し、 IR body filler が ignored 時にそれを true で構築、 `distinctPreviewsById` (runtime) で filter
- (C) IR body filler で ignored preview には sentinel id (`__cpl_ignored_<sourceFqn>`) を使い、 `distinctPreviewsById` で filter

redesign 内で対応するには各案とも非自明で、 V1 削除を遅らせるリスクと天秤にかけた結果 follow-up 送りとしました。 別 ticket での対応を予定しています。

File: compiler-plugin/src/main/kotlin/me/tbsten/compose/preview/lab/compiler/ir/HintDiscoveryV2.kt [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * - **filter 条件**: 関数名が `previewHint_` で始まること + 外部 module 由来であること
+ *   + 戻り値型が `CollectedPreview` であること (sanity check)
+ */
+internal fun discoverHintsV2(pluginContext: IrPluginContext, compatContext: CompatContext,): List<IrSimpleFunction> {
+    if (!compatContext.supportsKlibCrossModuleHint() && pluginContext.platform?.isJvm() != true) {
+        return emptyList()
+    }
+
+    val packageView = pluginContext.moduleDescriptor.getPackage(PreviewLabFirBuiltIns.HINT_PACKAGE_V2)
+    val functionNames = packageView.memberScope.getFunctionNames()
```
Comment (TBSten) [inline compiler-plugin/src/main/kotlin/me/tbsten/compose/preview/lab/compiler/ir/HintDiscoveryV2.kt:54]: ⚠️ **T05 (#173) でこの実装が根本的に書き換わります**: `moduleDescriptor.getPackage(...).memberScope.getFunctionNames()` は **外部 module (jar / klib) の宣言を on-demand load しない** ため、 cross-module hint discovery が実は機能していません。

このことが PR #171 時点で発覚しなかった理由は、 V1 (旧モジュール集約 hint) が同時に動いていて discovery を担当していたから。 T05 で V1 を削除した時点で発覚しました。

T05 で **fixed-name 方式** に変更:
```kotlin
val hintSymbols = pluginContext.referenceFunctions(
    CallableId(HINT_PACKAGE_V2, Name.identifier("previewHint"))
)
```
`referenceFunctions(CallableId)` は K2 で classpath 全体を walk するため cross-module でも発見可能 (V1 `previewLabExport(Marker)` pattern と同じ)。 hint generator 側も fixed-name + marker class 方式に変更されます (PR #173 description 参照)。

File: compiler-plugin/src/main/kotlin/me/tbsten/compose/preview/lab/compiler/ir/PreviewListIrBuilder.kt [hunk 4]
```diff
@@ -279,6 +302,15 @@ internal class PreviewListIrBuilder(
                     arguments[1] = dependencyValue
                 }
             }
```
Comment (TBSten) [inline compiler-plugin/src/main/kotlin/me/tbsten/compose/preview/lab/compiler/ir/PreviewListIrBuilder.kt:304]: ℹ️ **T05 (#173) でこのループの hint call が変わります**: 現状は `hint()` (引数なし) ですが、 T05 で hint 関数 signature が `previewHint(value: PreviewHintMarker_<hash>?): CollectedPreview` になるため、 `hint(null)` (marker arg に null) を渡す形になります。 marker arg は IdSignature 区別が目的で実値不要なので null で十分。

また、 T05 で V1 dep getter (`addAll(dependencyGetter())`) は削除されるので、 このブロック全体が `hints.forEach { add(hint(null)) }` だけになります。
